darwin: don't include cf-private in framework dependencies

LnL7 opened the pull request:
###### Motivation for this change

Using cf-private can cause problems, it's better to include it
explicitly in places where it's actually required.

Fixes #49781

NOTE: this will break some things, a bit more testing is required.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

LnL7 committed:
darwin: don't include cf-private in framework dependencies

Using cf-private can cause problems, it's better to include it
explicitly in places where it's actually required.

LnL7 committed:
SDL2: include cf-private on darwin

It's not included implicitly by the frameworks anymore.

    Undefined symbols for architecture x86_64:
      "_NSDefaultRunLoopMode", referenced from:
          _Cocoa_PumpEvents in SDL_cocoaevents.o
      "_NSURLIsAliasFileKey", referenced from:
          -[SDLWindow performDragOperation:] in SDL_cocoawindow.o
      "_OBJC_CLASS_$_NSArray", referenced from:
          objc-class-ref in SDL_cocoaclipboard.o
          objc-class-ref in SDL_cocoakeyboard.o
          objc-class-ref in SDL_cocoawindow.o
      "_OBJC_CLASS_$_NSData", referenced from:
          objc-class-ref in SDL_cocoamouse.o
      "_OBJC_CLASS_$_NSDate", referenced from:
          objc-class-ref in SDL_cocoaevents.o
      "_OBJC_CLASS_$_NSDictionary", referenced from:
          objc-class-ref in SDL_cocoaevents.o
      "_OBJC_CLASS_$_NSMutableArray", referenced from:
          objc-class-ref in SDL_cocoawindow.o
      "_OBJC_CLASS_$_NSURL", referenced from:
          objc-class-ref in SDL_cocoawindow.o
      "_OBJC_CLASS_$_NSUserDefaults", referenced from:
          objc-class-ref in SDL_cocoaevents.o
      "_OBJC_EHTYPE_$_NSException", referenced from:
          GCC_except_table67 in SDL_cocoawindow.o
    ld: symbol(s) not found for architecture x86_64

LnL7 committed:
SDL: include cf-private on darwin

It's not included implicitly by the frameworks anymore.

    Undefined symbols for architecture x86_64:
      "_NSDefaultRunLoopMode", referenced from:
          _QZ_PumpEvents in SDL_QuartzEvents.o
      "_OBJC_CLASS_$_NSArray", referenced from:
          objc-class-ref in SDL_QuartzEvents.o
      "_OBJC_CLASS_$_NSDate", referenced from:
          objc-class-ref in SDL_QuartzEvents.o
    ld: symbol(s) not found for architecture x86_64

LnL7 committed:
SDL: move overrides out of all-packages.nix

Using defaults makes it much easier to understand what the values are.

GrahamcOfBorg commented:
<!--REQUEST_ID=35553ac2-cc59-48b8-8e3c-306fc026a064-->
Success on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.49768&attempt_id=6d595bce-329a-4962-9e5e-c8917bb1b9b0)

Attempted: SDL, SDL2

<details><summary>Partial log (click to expand)</summary><p>

```
patching script interpreter paths in /nix/store/1706f3s31i1m72xycjg4lnjvwh10l2a2-SDL2-2.0.8
checking for references to /build in /nix/store/1706f3s31i1m72xycjg4lnjvwh10l2a2-SDL2-2.0.8...
shrinking RPATHs of ELF executables and libraries in /nix/store/5w0jysabsm92jbr93d2a0a4zv8amwp1h-SDL2-2.0.8-dev
strip is /nix/store/p9akxn2sfy4wkhqdqa3li97pc6jaz3r1-binutils-2.30/bin/strip
stripping (with command strip and flags -S) in /nix/store/5w0jysabsm92jbr93d2a0a4zv8amwp1h-SDL2-2.0.8-dev/lib  /nix/store/5w0jysabsm92jbr93d2a0a4zv8amwp1h-SDL2-2.0.8-dev/bin
patching script interpreter paths in /nix/store/5w0jysabsm92jbr93d2a0a4zv8amwp1h-SDL2-2.0.8-dev
/nix/store/5w0jysabsm92jbr93d2a0a4zv8amwp1h-SDL2-2.0.8-dev/bin/sdl2-config: interpreter directive changed from "/bin/sh" to "/nix/store/dsyc1z7ck08ga7l0b1jcxx35wj69qcii-bash-4.4-p23/bin/sh"
checking for references to /build in /nix/store/5w0jysabsm92jbr93d2a0a4zv8amwp1h-SDL2-2.0.8-dev...
/nix/store/ldqanhma1j9i73vscxif35hzgq2lk9k6-SDL-1.2.15
/nix/store/1706f3s31i1m72xycjg4lnjvwh10l2a2-SDL2-2.0.8
```
</p></details>

GrahamcOfBorg commented:
<!--REQUEST_ID=35553ac2-cc59-48b8-8e3c-306fc026a064-->
Success on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.49768&attempt_id=492a41ba-627d-4ec3-a245-94a864fce405)

Attempted: SDL, SDL2

<details><summary>Partial log (click to expand)</summary><p>

```
/nix/store/8fyy3q8w783n63k8q2kxs5hb73p3lzyn-SDL-1.2.15/lib/libSDL-1.2.0.dylib: fixing dylib
/nix/store/8fyy3q8w783n63k8q2kxs5hb73p3lzyn-SDL-1.2.15/lib/libSDL-1.2.0.dylib: fixing dylib
/nix/store/8fyy3q8w783n63k8q2kxs5hb73p3lzyn-SDL-1.2.15/lib/libSDL-1.2.0.dylib: fixing dylib
gzipping man pages under /nix/store/kpiicihqqj7dy19ln4kkbrph7wrys5hc-SDL-1.2.15-dev/share/man/
strip is /nix/store/g5r4apl0za012ffs6ladinwa5w0m1l3k-cctools-binutils-darwin/bin/strip
stripping (with command strip and flags -S) in /nix/store/kpiicihqqj7dy19ln4kkbrph7wrys5hc-SDL-1.2.15-dev/lib  /nix/store/kpiicihqqj7dy19ln4kkbrph7wrys5hc-SDL-1.2.15-dev/bin
patching script interpreter paths in /nix/store/kpiicihqqj7dy19ln4kkbrph7wrys5hc-SDL-1.2.15-dev
/nix/store/kpiicihqqj7dy19ln4kkbrph7wrys5hc-SDL-1.2.15-dev/bin/sdl-config: interpreter directive changed from "/bin/sh" to "/nix/store/n9hba031gjky8hpjgx9fnlaxhidyzxbz-bash-4.4-p23/bin/sh"
/nix/store/8fyy3q8w783n63k8q2kxs5hb73p3lzyn-SDL-1.2.15
/nix/store/qr49x1mndr48qayr6j9x9wm7l082qwg2-SDL2-2.0.8
```
</p></details>

LnL7 committed:
atomicparsley: include cf-private on darwin

    Undefined symbols for architecture x86_64:
      "_OBJC_CLASS_$_NSDictionary", referenced from:
          objc-class-ref in nsfile.o
          objc-class-ref in nsimage.o
    ld: symbol(s) not found for architecture x86_64

GrahamcOfBorg commented:
<!--REQUEST_ID=e43c0ced-560f-4f45-944e-343747ade94a-->
Success on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.49768&attempt_id=c8dd91ba-3adc-413d-a9db-de8b668de12e)

Attempted: SDL, SDL2, atomicparsley

<details><summary>Partial log (click to expand)</summary><p>

```
post-installation fixup
shrinking RPATHs of ELF executables and libraries in /nix/store/7vih3zfzylrzb37zdb3i6f6sh6hypdjd-atomicparsley-0.9.6
shrinking /nix/store/7vih3zfzylrzb37zdb3i6f6sh6hypdjd-atomicparsley-0.9.6/bin/AtomicParsley
strip is /nix/store/p9akxn2sfy4wkhqdqa3li97pc6jaz3r1-binutils-2.30/bin/strip
stripping (with command strip and flags -S) in /nix/store/7vih3zfzylrzb37zdb3i6f6sh6hypdjd-atomicparsley-0.9.6/bin
patching script interpreter paths in /nix/store/7vih3zfzylrzb37zdb3i6f6sh6hypdjd-atomicparsley-0.9.6
checking for references to /build in /nix/store/7vih3zfzylrzb37zdb3i6f6sh6hypdjd-atomicparsley-0.9.6...
/nix/store/ldqanhma1j9i73vscxif35hzgq2lk9k6-SDL-1.2.15
/nix/store/1706f3s31i1m72xycjg4lnjvwh10l2a2-SDL2-2.0.8
/nix/store/7vih3zfzylrzb37zdb3i6f6sh6hypdjd-atomicparsley-0.9.6
```
</p></details>

GrahamcOfBorg commented:
<!--REQUEST_ID=e43c0ced-560f-4f45-944e-343747ade94a-->
Success on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.49768&attempt_id=101f33fc-37fe-4b4a-b33f-9e59e434f50c)

Attempted: SDL, SDL2, atomicparsley

<details><summary>Partial log (click to expand)</summary><p>

```
post-installation fixup
shrinking RPATHs of ELF executables and libraries in /nix/store/n2j1k002bvvj1vkzm1aapjxad61aaspr-atomicparsley-0.9.6
shrinking /nix/store/n2j1k002bvvj1vkzm1aapjxad61aaspr-atomicparsley-0.9.6/bin/AtomicParsley
strip is /nix/store/vcc4svb8gy29g4pam2zja6llkbcwsyiq-binutils-2.30/bin/strip
stripping (with command strip and flags -S) in /nix/store/n2j1k002bvvj1vkzm1aapjxad61aaspr-atomicparsley-0.9.6/bin
patching script interpreter paths in /nix/store/n2j1k002bvvj1vkzm1aapjxad61aaspr-atomicparsley-0.9.6
checking for references to /build in /nix/store/n2j1k002bvvj1vkzm1aapjxad61aaspr-atomicparsley-0.9.6...
/nix/store/w6ayvanz3bjnydyqsnmdpgckq5a5hqz4-SDL-1.2.15
/nix/store/w5g8gcivvvi9binhmy6i34gi5b9g48yn-SDL2-2.0.8
/nix/store/n2j1k002bvvj1vkzm1aapjxad61aaspr-atomicparsley-0.9.6
```
</p></details>

GrahamcOfBorg commented:
<!--REQUEST_ID=e43c0ced-560f-4f45-944e-343747ade94a-->
Success on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.49768&attempt_id=dacb5ed9-346a-4a50-a7a2-23857e245ac0)

Attempted: SDL, SDL2, atomicparsley

<details><summary>Partial log (click to expand)</summary><p>

```
/nix/store/8fyy3q8w783n63k8q2kxs5hb73p3lzyn-SDL-1.2.15/lib/libSDL-1.2.0.dylib: fixing dylib
/nix/store/8fyy3q8w783n63k8q2kxs5hb73p3lzyn-SDL-1.2.15/lib/libSDL-1.2.0.dylib: fixing dylib
gzipping man pages under /nix/store/kpiicihqqj7dy19ln4kkbrph7wrys5hc-SDL-1.2.15-dev/share/man/
strip is /nix/store/g5r4apl0za012ffs6ladinwa5w0m1l3k-cctools-binutils-darwin/bin/strip
stripping (with command strip and flags -S) in /nix/store/kpiicihqqj7dy19ln4kkbrph7wrys5hc-SDL-1.2.15-dev/lib  /nix/store/kpiicihqqj7dy19ln4kkbrph7wrys5hc-SDL-1.2.15-dev/bin
patching script interpreter paths in /nix/store/kpiicihqqj7dy19ln4kkbrph7wrys5hc-SDL-1.2.15-dev
/nix/store/kpiicihqqj7dy19ln4kkbrph7wrys5hc-SDL-1.2.15-dev/bin/sdl-config: interpreter directive changed from "/bin/sh" to "/nix/store/n9hba031gjky8hpjgx9fnlaxhidyzxbz-bash-4.4-p23/bin/sh"
/nix/store/8fyy3q8w783n63k8q2kxs5hb73p3lzyn-SDL-1.2.15
/nix/store/qr49x1mndr48qayr6j9x9wm7l082qwg2-SDL2-2.0.8
/nix/store/n3vmdnxpla19ljl7h33g13hvnlqwqlww-atomicparsley-0.9.6
```
</p></details>

LnL7 committed:
pinentry_mac: add cf-private

    Undefined symbols for architecture x86_64:
      "_OBJC_CLASS_$_NSArray", referenced from:
          objc-class-ref in GPGDefaults.o
      "_OBJC_CLASS_$_NSDictionary", referenced from:
          objc-class-ref in PinentryController.o
          objc-class-ref in GPGDefaults.o
          objc-class-ref in KeychainSupport.o
      "_OBJC_CLASS_$_NSMutableDictionary", referenced from:
          objc-class-ref in GPGDefaults.o
      "_OBJC_CLASS_$_NSSet", referenced from:
          objc-class-ref in GPGDefaults.o
      "_OBJC_CLASS_$_NSUserDefaults", referenced from:
          objc-class-ref in GPGDefaults.o
    ld: symbol(s) not found for architecture x86_64

LnL7 committed:
vim: add cf-private on darwin

    Undefined symbols for architecture x86_64:
      "_OBJC_CLASS_$_NSArray", referenced from:
          objc-class-ref in os_macosx.o
    ld: symbol(s) not found for architecture x86_64

LnL7 committed:
contacts: add cf-private

    Undefined symbols for architecture x86_64:
      "_OBJC_CLASS_$_NSArray", referenced from:
          objc-class-ref in contacts.o
          objc-class-ref in FormatHelper.o
      "_OBJC_CLASS_$_NSMutableArray", referenced from:
          objc-class-ref in FormatHelper.o
    ld: symbol(s) not found for architecture x86_64

GrahamcOfBorg commented:
<!--REQUEST_ID=a60c946e-1dc1-4a37-8bef-c5d8ae5554d3-->
Success on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.49768&attempt_id=9ee682ba-0fa0-4851-8b54-8f5c99f93304)

Attempted: SDL, SDL2, atomicparsley, vim

The following builds were skipped because they don't evaluate on aarch64-linux: pinentry_mac

<details><summary>Partial log (click to expand)</summary><p>

```
to ~/.config/nixpkgs/config.nix.


these paths will be fetched (6.36 MiB download, 30.54 MiB unpacked):
  /nix/store/004qkskyals3hc6yy6y6c6wvi0vxzvcb-vim-8.1.0490
copying path '/nix/store/004qkskyals3hc6yy6y6c6wvi0vxzvcb-vim-8.1.0490' from 'https://cache.nixos.org'...
/nix/store/ldqanhma1j9i73vscxif35hzgq2lk9k6-SDL-1.2.15
/nix/store/1706f3s31i1m72xycjg4lnjvwh10l2a2-SDL2-2.0.8
/nix/store/7vih3zfzylrzb37zdb3i6f6sh6hypdjd-atomicparsley-0.9.6
/nix/store/004qkskyals3hc6yy6y6c6wvi0vxzvcb-vim-8.1.0490
```
</p></details>

GrahamcOfBorg commented:
<!--REQUEST_ID=a60c946e-1dc1-4a37-8bef-c5d8ae5554d3-->
Success on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.49768&attempt_id=672fceb9-e6bd-48db-9a8b-55b3066ca42c)

Attempted: SDL, SDL2, atomicparsley, pinentry_mac, vim

<details><summary>Partial log (click to expand)</summary><p>

```
basename: invalid option -- 'w'
Try 'basename --help' for more information.
/nix/store/31h560p5ia8wv2lkp026lar8mpkdbm21-vim-8.1.0490/share/vim/vim81/tools/ref: interpreter directive changed from "/bin/sh" to "/nix/store/n9hba031gjky8hpjgx9fnlaxhidyzxbz-bash-4.4-p23/bin/sh"
/nix/store/31h560p5ia8wv2lkp026lar8mpkdbm21-vim-8.1.0490/share/vim/vim81/tools/vimm: interpreter directive changed from "/bin/sh" to "/nix/store/n9hba031gjky8hpjgx9fnlaxhidyzxbz-bash-4.4-p23/bin/sh"
/nix/store/31h560p5ia8wv2lkp026lar8mpkdbm21-vim-8.1.0490/share/vim/vim81/tools/vimspell.sh: interpreter directive changed from "/bin/sh" to "/nix/store/n9hba031gjky8hpjgx9fnlaxhidyzxbz-bash-4.4-p23/bin/sh"
/nix/store/8fyy3q8w783n63k8q2kxs5hb73p3lzyn-SDL-1.2.15
/nix/store/qr49x1mndr48qayr6j9x9wm7l082qwg2-SDL2-2.0.8
/nix/store/n3vmdnxpla19ljl7h33g13hvnlqwqlww-atomicparsley-0.9.6
/nix/store/2ip8n4f0r8rxaylq0ca4z21l7dlml948-pinentry-mac-0.9.4
/nix/store/31h560p5ia8wv2lkp026lar8mpkdbm21-vim-8.1.0490
```
</p></details>

GrahamcOfBorg commented:
<!--REQUEST_ID=a60c946e-1dc1-4a37-8bef-c5d8ae5554d3-->
Success on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.49768&attempt_id=17a93a47-0b03-4791-934e-fe422eaaca2f)

Attempted: SDL, SDL2, atomicparsley, vim

The following builds were skipped because they don't evaluate on x86_64-linux: pinentry_mac

<details><summary>Partial log (click to expand)</summary><p>

```
shrinking RPATHs of ELF executables and libraries in /nix/store/p3q7il15fakcgmmql0df0ywrg1lrqknr-SDL2-2.0.8-dev
strip is /nix/store/vcc4svb8gy29g4pam2zja6llkbcwsyiq-binutils-2.30/bin/strip
stripping (with command strip and flags -S) in /nix/store/p3q7il15fakcgmmql0df0ywrg1lrqknr-SDL2-2.0.8-dev/lib  /nix/store/p3q7il15fakcgmmql0df0ywrg1lrqknr-SDL2-2.0.8-dev/bin
patching script interpreter paths in /nix/store/p3q7il15fakcgmmql0df0ywrg1lrqknr-SDL2-2.0.8-dev
/nix/store/p3q7il15fakcgmmql0df0ywrg1lrqknr-SDL2-2.0.8-dev/bin/sdl2-config: interpreter directive changed from "/bin/sh" to "/nix/store/r47p5pzx52m3n34vdgqpk5rvqgm0m24m-bash-4.4-p23/bin/sh"
checking for references to /build in /nix/store/p3q7il15fakcgmmql0df0ywrg1lrqknr-SDL2-2.0.8-dev...
/nix/store/w6ayvanz3bjnydyqsnmdpgckq5a5hqz4-SDL-1.2.15
/nix/store/w5g8gcivvvi9binhmy6i34gi5b9g48yn-SDL2-2.0.8
/nix/store/n2j1k002bvvj1vkzm1aapjxad61aaspr-atomicparsley-0.9.6
/nix/store/2p3nfi1xhiy4l3j6mw01zpyy5qjjvw7i-vim-8.1.0490
```
</p></details>

LnL7 committed:
emacs: include cf-private on darwin and don't propagate frameworks

    Undefined symbols for architecture x86_64:
      "_CFNotificationCenterAddObserver", referenced from:
          _macfont_copy_available_families_cache in macfont.o
      "_CFNotificationCenterGetLocalCenter", referenced from:
          _macfont_copy_available_families_cache in macfont.o
      "_NSDefaultRunLoopMode", referenced from:
          _ns_send_appdefined in nsterm.o
          -[EmacsApp run] in nsterm.o
      "_OBJC_CLASS_$_NSArray", referenced from:
          objc-class-ref in nsterm.o
          objc-class-ref in nsmenu.o
          objc-class-ref in nsselect.o
      "_OBJC_CLASS_$_NSData", referenced from:
          objc-class-ref in nsimage.o
      "_OBJC_CLASS_$_NSDate", referenced from:
          objc-class-ref in nsterm.o
      "_OBJC_CLASS_$_NSDictionary", referenced from:
          objc-class-ref in macfont.o
      "_OBJC_CLASS_$_NSLocale", referenced from:
          objc-class-ref in nsterm.o
      "_OBJC_CLASS_$_NSMutableArray", referenced from:
          objc-class-ref in nsterm.o
          objc-class-ref in nsmenu.o
      "_OBJC_CLASS_$_NSMutableDictionary", referenced from:
          objc-class-ref in nsmenu.o
          objc-class-ref in nsselect.o
      "_OBJC_CLASS_$_NSMutableSet", referenced from:
          objc-class-ref in nsterm.o
      "_OBJC_CLASS_$_NSRunLoop", referenced from:
          objc-class-ref in nsmenu.o
      "_OBJC_CLASS_$_NSTimer", referenced from:
          objc-class-ref in nsterm.o
          objc-class-ref in nsmenu.o
      "_OBJC_CLASS_$_NSURL", referenced from:
          objc-class-ref in nsterm.o
          objc-class-ref in nsfns.o
      "_OBJC_CLASS_$_NSUserDefaults", referenced from:
          objc-class-ref in nsterm.o
          objc-class-ref in nsfns.o
      "_OBJC_EHTYPE_$_NSException", referenced from:
          GCC_except_table8 in nsterm.o
          GCC_except_table1 in nsselect.o
    ld: symbol(s) not found for architecture x86_64

GrahamcOfBorg commented:
<!--REQUEST_ID=8aaaab54-6226-41e8-991e-fb5ccaa7ea73-->
Success on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.49768&attempt_id=4c8f395b-5adc-4f64-addb-48d07b3dabf1)

Attempted: SDL, SDL2, atomicparsley, vim

The following builds were skipped because they don't evaluate on aarch64-linux: contacts, pinentry_mac

<details><summary>Partial log (click to expand)</summary><p>

```

b) For `nix-env`, `nix-build`, `nix-shell` or any other Nix command you can add
  { allowUnsupportedSystem = true; }
to ~/.config/nixpkgs/config.nix.


/nix/store/ldqanhma1j9i73vscxif35hzgq2lk9k6-SDL-1.2.15
/nix/store/1706f3s31i1m72xycjg4lnjvwh10l2a2-SDL2-2.0.8
/nix/store/7vih3zfzylrzb37zdb3i6f6sh6hypdjd-atomicparsley-0.9.6
/nix/store/004qkskyals3hc6yy6y6c6wvi0vxzvcb-vim-8.1.0490
```
</p></details>

GrahamcOfBorg commented:
<!--REQUEST_ID=8aaaab54-6226-41e8-991e-fb5ccaa7ea73-->
Success on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.49768&attempt_id=26bdba95-af72-4bf9-88e1-53b97b261720)

Attempted: SDL, SDL2, atomicparsley, contacts, pinentry_mac, vim

<details><summary>Partial log (click to expand)</summary><p>

```
post-installation fixup
strip is /nix/store/jicg9d0a282yzcig82308jf016iy4qk7-Toolchains/XcodeDefault.xctoolchain/bin/strip
stripping (with command strip and flags -S) in /nix/store/3b58rj8srw94gqfx4lcjsrmv18vy8v97-contacts-1.1a-3/bin
patching script interpreter paths in /nix/store/3b58rj8srw94gqfx4lcjsrmv18vy8v97-contacts-1.1a-3
/nix/store/8fyy3q8w783n63k8q2kxs5hb73p3lzyn-SDL-1.2.15
/nix/store/qr49x1mndr48qayr6j9x9wm7l082qwg2-SDL2-2.0.8
/nix/store/n3vmdnxpla19ljl7h33g13hvnlqwqlww-atomicparsley-0.9.6
/nix/store/3b58rj8srw94gqfx4lcjsrmv18vy8v97-contacts-1.1a-3
/nix/store/2ip8n4f0r8rxaylq0ca4z21l7dlml948-pinentry-mac-0.9.4
/nix/store/31h560p5ia8wv2lkp026lar8mpkdbm21-vim-8.1.0490
```
</p></details>

GrahamcOfBorg commented:
<!--REQUEST_ID=8aaaab54-6226-41e8-991e-fb5ccaa7ea73-->
Success on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.49768&attempt_id=bd6e2532-1e6a-4b34-a994-c6510fd5253b)

Attempted: SDL, SDL2, atomicparsley, vim

The following builds were skipped because they don't evaluate on x86_64-linux: contacts, pinentry_mac

<details><summary>Partial log (click to expand)</summary><p>

```

b) For `nix-env`, `nix-build`, `nix-shell` or any other Nix command you can add
  { allowUnsupportedSystem = true; }
to ~/.config/nixpkgs/config.nix.


/nix/store/w6ayvanz3bjnydyqsnmdpgckq5a5hqz4-SDL-1.2.15
/nix/store/w5g8gcivvvi9binhmy6i34gi5b9g48yn-SDL2-2.0.8
/nix/store/n2j1k002bvvj1vkzm1aapjxad61aaspr-atomicparsley-0.9.6
/nix/store/2p3nfi1xhiy4l3j6mw01zpyy5qjjvw7i-vim-8.1.0490
```
</p></details>

GrahamcOfBorg commented:
<!--REQUEST_ID=45a81fc0-2720-4175-b5d5-2351e7164279-->
Success on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.49768&attempt_id=e39e6db8-6c04-43bd-88d0-0b49e2ed2485)

Attempted: SDL, SDL2, atomicparsley, emacs, vim

The following builds were skipped because they don't evaluate on x86_64-linux: contacts, pinentry_mac

<details><summary>Partial log (click to expand)</summary><p>

```
copying path '/nix/store/ivmry7bh8ihiw1xqzzrb9fg1n2f5k5k0-m17n-lib-1.8.0' from 'https://cache.nixos.org'...
copying path '/nix/store/ar0k6fpylx2skacpj5ki45r6ni3q97vj-gdk-pixbuf-2.36.12' from 'https://cache.nixos.org'...
copying path '/nix/store/p4b0avnpcvvj6b108zba9j61fgir5b2k-gtk+3-3.22.30' from 'https://cache.nixos.org'...
copying path '/nix/store/x3kyz5r5xdnv9d7qqfpy6f750qshjp97-librsvg-2.42.4' from 'https://cache.nixos.org'...
copying path '/nix/store/4vcdiljhrsnl73yabsn56n0nprxf6g20-emacs-26.1' from 'https://cache.nixos.org'...
/nix/store/w6ayvanz3bjnydyqsnmdpgckq5a5hqz4-SDL-1.2.15
/nix/store/w5g8gcivvvi9binhmy6i34gi5b9g48yn-SDL2-2.0.8
/nix/store/n2j1k002bvvj1vkzm1aapjxad61aaspr-atomicparsley-0.9.6
/nix/store/4vcdiljhrsnl73yabsn56n0nprxf6g20-emacs-26.1
/nix/store/2p3nfi1xhiy4l3j6mw01zpyy5qjjvw7i-vim-8.1.0490
```
</p></details>

GrahamcOfBorg commented:
<!--REQUEST_ID=45a81fc0-2720-4175-b5d5-2351e7164279-->
Success on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.49768&attempt_id=984df8ed-fbce-4954-ac60-f3990a0d7fab)

Attempted: SDL, SDL2, atomicparsley, contacts, emacs, pinentry_mac, vim

<details><summary>Partial log (click to expand)</summary><p>

```
stripping (with command strip and flags -S) in /nix/store/932r74r2m5yav4177bkl3nq69gj8jvfk-emacs-26.1/lib  /nix/store/932r74r2m5yav4177bkl3nq69gj8jvfk-emacs-26.1/libexec  /nix/store/932r74r2m5yav4177bkl3nq69gj8jvfk-emacs-26.1/bin
patching script interpreter paths in /nix/store/932r74r2m5yav4177bkl3nq69gj8jvfk-emacs-26.1
/nix/store/932r74r2m5yav4177bkl3nq69gj8jvfk-emacs-26.1/libexec/emacs/26.1/x86_64-apple-darwin16.3.0/rcs2log: interpreter directive changed from " /bin/sh" to "/nix/store/n9hba031gjky8hpjgx9fnlaxhidyzxbz-bash-4.4-p23/bin/sh"
/nix/store/8fyy3q8w783n63k8q2kxs5hb73p3lzyn-SDL-1.2.15
/nix/store/qr49x1mndr48qayr6j9x9wm7l082qwg2-SDL2-2.0.8
/nix/store/n3vmdnxpla19ljl7h33g13hvnlqwqlww-atomicparsley-0.9.6
/nix/store/3b58rj8srw94gqfx4lcjsrmv18vy8v97-contacts-1.1a-3
/nix/store/932r74r2m5yav4177bkl3nq69gj8jvfk-emacs-26.1
/nix/store/2ip8n4f0r8rxaylq0ca4z21l7dlml948-pinentry-mac-0.9.4
/nix/store/31h560p5ia8wv2lkp026lar8mpkdbm21-vim-8.1.0490
```
</p></details>

GrahamcOfBorg commented:
<!--REQUEST_ID=45a81fc0-2720-4175-b5d5-2351e7164279-->
Success on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.49768&attempt_id=a54244cc-7704-49ca-846c-25e67e26af33)

Attempted: SDL, SDL2, atomicparsley, emacs, vim

The following builds were skipped because they don't evaluate on aarch64-linux: contacts, pinentry_mac

<details><summary>Partial log (click to expand)</summary><p>

```
shrinking /nix/store/nc277chky7jm7wajmns0b5s4g3vvl5lz-emacs-26.1/bin/.ctags-wrapped
shrinking /nix/store/nc277chky7jm7wajmns0b5s4g3vvl5lz-emacs-26.1/bin/.emacsclient-wrapped
shrinking /nix/store/nc277chky7jm7wajmns0b5s4g3vvl5lz-emacs-26.1/bin/.ebrowse-wrapped
shrinking /nix/store/nc277chky7jm7wajmns0b5s4g3vvl5lz-emacs-26.1/bin/.emacs-26.1-wrapped
gzipping man pages under /nix/store/nc277chky7jm7wajmns0b5s4g3vvl5lz-emacs-26.1/share/man/
strip is /nix/store/p9akxn2sfy4wkhqdqa3li97pc6jaz3r1-binutils-2.30/bin/strip
stripping (with command strip and flags -S) in /nix/store/nc277chky7jm7wajmns0b5s4g3vvl5lz-emacs-26.1/lib  /nix/store/nc277chky7jm7wajmns0b5s4g3vvl5lz-emacs-26.1/libexec  /nix/store/nc277chky7jm7wajmns0b5s4g3vvl5lz-emacs-26.1/bin
patching script interpreter paths in /nix/store/nc277chky7jm7wajmns0b5s4g3vvl5lz-emacs-26.1
/nix/store/nc277chky7jm7wajmns0b5s4g3vvl5lz-emacs-26.1/libexec/emacs/26.1/aarch64-unknown-linux-gnu/.rcs2log-wrapped: interpreter directive changed from " /bin/sh" to "/nix/store/dsyc1z7ck08ga7l0b1jcxx35wj69qcii-bash-4.4-p23/bin/sh"
checking for references to /build in /nix/store/nc277chky7jm7wajmns0b5s4g3vvl5lz-emacs-26.1...
```
</p></details>

LnL7 committed:
quartz-wm: add cf-private

    Undefined symbols for architecture x86_64:
      "_CFNotificationCenterAddObserver", referenced from:
          _main in main.o
      "_CFNotificationCenterGetDistributedCenter", referenced from:
          _main in main.o
      "_OBJC_CLASS_$_NSTimer", referenced from:
          objc-class-ref in main.o
          objc-class-ref in x-screen.o
      "_OBJC_EHTYPE_$_NSException", referenced from:
          GCC_except_table25 in main.o
    ld: symbol(s) not found for architecture x86_64

LnL7 committed:
xquartz: add cf-private

    Undefined symbols for architecture x86_64:
      "_NSDefaultRunLoopMode", referenced from:
          _X11ApplicationMain in libXquartz.a(X11Application.o)
      "_OBJC_CLASS_$_NSArray", referenced from:
          objc-class-ref in libXquartz.a(X11Application.o)
          objc-class-ref in libXquartz.a(X11Controller.o)
          objc-class-ref in libxpbproxy.a(x-selection.o)
      "_OBJC_CLASS_$_NSData", referenced from:
          objc-class-ref in libxpbproxy.a(x-selection.o)
      "_OBJC_CLASS_$_NSDictionary", referenced from:
          objc-class-ref in libxpbproxy.a(x-selection.o)
      "_OBJC_CLASS_$_NSMutableArray", referenced from:
          objc-class-ref in libXquartz.a(X11Application.o)
          objc-class-ref in libXquartz.a(X11Controller.o)
      "_OBJC_CLASS_$_NSMutableDictionary", referenced from:
          objc-class-ref in libXquartz.a(X11Application.o)
      "_OBJC_CLASS_$_NSRunLoop", referenced from:
          objc-class-ref in libXquartz.a(X11Application.o)
      "_OBJC_CLASS_$_NSURL", referenced from:
          objc-class-ref in libXquartz.a(X11Application.o)
      "_OBJC_EHTYPE_$_NSException", referenced from:
          GCC_except_table29 in libxpbproxy.a(x-selection.o)
    ld: symbol(s) not found for architecture x86_64

LnL7 committed:
xorg.xorgserver: add cf-private on darwin

    Undefined symbols for architecture x86_64:
      "_NSDefaultRunLoopMode", referenced from:
          _X11ApplicationMain in libXquartz.a(X11Application.o)
      "_OBJC_CLASS_$_NSArray", referenced from:
          objc-class-ref in libXquartz.a(X11Application.o)
          objc-class-ref in libXquartz.a(X11Controller.o)
          objc-class-ref in libxpbproxy.a(x-selection.o)
      "_OBJC_CLASS_$_NSData", referenced from:
          objc-class-ref in libxpbproxy.a(x-selection.o)
      "_OBJC_CLASS_$_NSDictionary", referenced from:
          objc-class-ref in libxpbproxy.a(x-selection.o)
      "_OBJC_CLASS_$_NSMutableArray", referenced from:
          objc-class-ref in libXquartz.a(X11Application.o)
          objc-class-ref in libXquartz.a(X11Controller.o)
      "_OBJC_CLASS_$_NSMutableDictionary", referenced from:
          objc-class-ref in libXquartz.a(X11Application.o)
      "_OBJC_CLASS_$_NSRunLoop", referenced from:
          objc-class-ref in libXquartz.a(X11Application.o)
      "_OBJC_CLASS_$_NSURL", referenced from:
          objc-class-ref in libXquartz.a(X11Application.o)
      "_OBJC_EHTYPE_$_NSException", referenced from:
          GCC_except_table29 in libxpbproxy.a(x-selection.o)
    ld: symbol(s) not found for architecture x86_64

GrahamcOfBorg commented:
<!--REQUEST_ID=a61cbfdc-4bd2-423c-be5a-de38e9dcf5b8-->
Success on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.49768&attempt_id=f932d39a-a4ea-4a1a-83c3-2abd0476b59e)

Attempted: SDL, SDL2, atomicparsley, emacs, vim, xorg.xorgserver

The following builds were skipped because they don't evaluate on x86_64-linux: contacts, pinentry_mac, quartz-wm, xquartz

<details><summary>Partial log (click to expand)</summary><p>

```
patching script interpreter paths in /nix/store/l9k4dhdipc4zxmn2xqki97jidwvmk4jq-SDL2-2.0.9-dev
/nix/store/l9k4dhdipc4zxmn2xqki97jidwvmk4jq-SDL2-2.0.9-dev/bin/sdl2-config: interpreter directive changed from "/bin/sh" to "/nix/store/r47p5pzx52m3n34vdgqpk5rvqgm0m24m-bash-4.4-p23/bin/sh"
checking for references to /build in /nix/store/l9k4dhdipc4zxmn2xqki97jidwvmk4jq-SDL2-2.0.9-dev...
copying path '/nix/store/4vcdiljhrsnl73yabsn56n0nprxf6g20-emacs-26.1' from 'https://cache.nixos.org'...
/nix/store/w6ayvanz3bjnydyqsnmdpgckq5a5hqz4-SDL-1.2.15
/nix/store/jlif2032zlf3r79a3fdrmgp3cqb16fqn-SDL2-2.0.9
/nix/store/n2j1k002bvvj1vkzm1aapjxad61aaspr-atomicparsley-0.9.6
/nix/store/4vcdiljhrsnl73yabsn56n0nprxf6g20-emacs-26.1
/nix/store/2p3nfi1xhiy4l3j6mw01zpyy5qjjvw7i-vim-8.1.0490
/nix/store/rx98jlf1vp4hmlpgjvpwqjcsnijz8nsg-xorg-server-1.19.6
```
</p></details>

GrahamcOfBorg commented:
<!--REQUEST_ID=a61cbfdc-4bd2-423c-be5a-de38e9dcf5b8-->
Success on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.49768&attempt_id=cf2312de-5095-419e-bccb-7ef56d1de55c)

Attempted: SDL, SDL2, atomicparsley, contacts, emacs, pinentry_mac, quartz-wm, vim, xorg.xorgserver, xquartz

<details><summary>Partial log (click to expand)</summary><p>

```
/nix/store/8fyy3q8w783n63k8q2kxs5hb73p3lzyn-SDL-1.2.15
/nix/store/d01ir5w0as1xc29zskgx0hrzlr4izwvb-SDL2-2.0.9
/nix/store/n3vmdnxpla19ljl7h33g13hvnlqwqlww-atomicparsley-0.9.6
/nix/store/3b58rj8srw94gqfx4lcjsrmv18vy8v97-contacts-1.1a-3
/nix/store/932r74r2m5yav4177bkl3nq69gj8jvfk-emacs-26.1
/nix/store/2ip8n4f0r8rxaylq0ca4z21l7dlml948-pinentry-mac-0.9.4
/nix/store/mnk9mx5s2p5rdrlzi4zmy9vvk5078znh-quartz-wm-1.3.1
/nix/store/31h560p5ia8wv2lkp026lar8mpkdbm21-vim-8.1.0490
/nix/store/03v0g1mw6fjzc9vwamdi86a00ipiy801-xorg-server-1.18.4
/nix/store/vf77j4vimcdg2vz5rjcsz45czm1gvg4i-xquartz-1.18.4
```
</p></details>

LnL7 commented:
I think this is mostly ready now, I'll keep an eye on hydra.

LnL7 commented:
Most of the regressions should be fixed. I'll try to look into adding the missing parts to https://github.com/apple/swift-corelibs-foundation, then we can get rid of those cf-private workarounds.
